daemon/algod/api/server/v2/dryrun.go

```diff
@@ -287,8 +289,17 @@ func (dl *dryrunLedger) Get(addr basics.Address, withPendingRewards bool) (basic
 	if ok {
 		any = true
 		app := dl.dr.Apps[appi]
-		out.AppParams = make(map[basics.AppIndex]basics.AppParams)
-		out.AppParams[basics.AppIndex(app.Id)] = ApplicationParamsToAppParams(&app.Params)
+		params := ApplicationParamsToAppParams(&app.Params)
+		if out.AppParams == nil {
+			out.AppParams = make(map[basics.AppIndex]basics.AppParams)
+			out.AppParams[basics.AppIndex(app.Id)] = params
+		} else {
+			ap, ok := out.AppParams[basics.AppIndex(app.Id)]
+			if ok {
+				MergeAppParams(&ap, &params)
```

Comment:
this fixes the situation when AccountData has AppParams with schema and predefined global state but a program to run is supplied as source with in DryrunRequest.Sources